src/plugins/compression.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Compression Library Wrapper
+ *
+ *  Created on: 08 Nov 2018
+ *      Author: Oliver Fernandes
+ */
+
+#ifndef SRC_PLUGINS_COMPRESSION_H_
+#define SRC_PLUGINS_COMPRESSION_H_
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <vector>
+
+#ifdef ENABLE_LZ4
+#include "lz4.h"
+#endif
+
+using ByteIterator = std::vector<char>::iterator;
+
+enum CompressionError {
+	COMP_SUCCESS = 0,
+	COMP_ERR_COMPRESSION_FAILED = 200,
+	COMP_ERR_DECOMPRESSION_FAILED = 300
+};
+
+class Compression {
+public:
+	/**
+	 * Compresses a series of bytes.
+     * Call to compress data.
+	 * @param[in] uncompressedStart Iterator pointing to the start of the array to be compressed
+	 * @param[in] uncompressedEnd Iterator pointing to the element past the last element of the array to be compressed
+	 * @param[in] compressed A std::vector<char> holding the compressed result. The vector will be appropriately resized.
+     *                       Any previous contents will be destroyed.
+	 */
+    virtual int compress(ByteIterator uncompressedStart, ByteIterator uncompressedEnd, std::vector<char>& compressed) = 0;
+	/**
+	 * Decompresses a series of bytes.
+     * Call to decompress data.
+	 * @param[in] compressedStart Iterator pointing to the start of the array to be decompressed
+	 * @param[in] compressedEnd Iterator pointing to the element past the last element of the array to be decompressed
+	 * @param[in] decompressed A std::vector<char> holding the decompressed result. The vector will be appropriately resized.
+     *                         Any previous contents will be destroyed. The first sizeof(size_t) bytes hold the uncompressed
+	 *                         size.
+	 */
```

Comment:
`@return` is missing in documentation

Reply:
probably an error index?

Reply:
See comment above. Return value helps when errors occur, can be ignored otherwise. Added appropriate docs.